### PR TITLE
core/slice: make slice_freezer_action() return 0 if freezing state is unchanged

### LINKED ISSUE #23278: systemd crashes somewhere in `bus_unit_method_freezer_generic`
It was found with `dfuzzer`

```sh
#0  0x00007fc3bdebe7ab in kill () at ../sysdeps/unix/syscall-template.S:120
#1  0x0000000000425e1d in crash (sig=6, siginfo=0x7ffcb71c3370, context=0x7ffcb71c3240) at ../src/core/crash-handler.c:83
#2  <signal handler called>
#3  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#4  0x00007fc3bdf37603 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#5  0x00007fc3bdeead86 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#6  0x00007fc3bdebd833 in __GI_abort () at abort.c:79
#7  0x00007fc3bf81d6f5 in log_assert_failed (text=0x7fc3c1447760 "!u->pending_freezer_message", file=0x7fc3c1445fc3 "src/core/dbus-unit.c", line=800,
    func=0x7fc3c144c9c0 <__PRETTY_FUNCTION__.60> "bus_unit_method_freezer_generic") at ../src/basic/log.c:866
#8  0x00007fc3c1046702 in bus_unit_method_freezer_generic (message=0x618000a37480, userdata=0x61b000011f80, error=0x7fc3b9c54740, action=FREEZER_THAW) at ../src/core/dbus-unit.c:800
#9  0x00007fc3c10467cf in bus_unit_method_thaw (message=0x618000a37480, userdata=0x61b000011f80, error=0x7fc3b9c54740) at ../src/core/dbus-unit.c:816
#10 0x00007fc3bf9a47ae in method_callbacks_run (bus=0x61c000005080, m=0x618000a37480, c=0x604000138c90, require_fallback=true, found_object=0x7fc3b9dac430) at ../src/libsystemd/sd-bus/bus-objects.c:406
#11 0x00007fc3bf9adfae in object_find_and_run (bus=0x61c000005080, m=0x618000a37480, p=0x604000310ad0 "/org/freedesktop/systemd1/unit", require_fallback=true, found_object=0x7fc3b9dac430)
    at ../src/libsystemd/sd-bus/bus-objects.c:1310
#12 0x00007fc3bf9af8f8 in bus_process_object (bus=0x61c000005080, m=0x618000a37480) at ../src/libsystemd/sd-bus/bus-objects.c:1440
#13 0x00007fc3bfa0ea97 in process_message (bus=0x61c000005080, m=0x618000a37480) at ../src/libsystemd/sd-bus/sd-bus.c:2962
#14 0x00007fc3bfa0efae in process_running (bus=0x61c000005080, ret=0x0) at ../src/libsystemd/sd-bus/sd-bus.c:3004
#15 0x00007fc3bfa11e0f in bus_process_internal (bus=0x61c000005080, ret=0x0) at ../src/libsystemd/sd-bus/sd-bus.c:3224
#16 0x00007fc3bfa11fa9 in sd_bus_process (bus=0x61c000005080, ret=0x0) at ../src/libsystemd/sd-bus/sd-bus.c:3251
#17 0x00007fc3bfa156aa in io_callback (s=0x6120000271c0, fd=24, revents=1, userdata=0x61c000005080) at ../src/libsystemd/sd-bus/sd-bus.c:3602
#18 0x00007fc3bfbcd720 in source_dispatch (s=0x6120000271c0) at ../src/libsystemd/sd-event/sd-event.c:3591
#19 0x00007fc3bfbd5f37 in sd_event_dispatch (e=0x617000000400) at ../src/libsystemd/sd-event/sd-event.c:4175
#20 0x00007fc3bfbd709b in sd_event_run (e=0x617000000400, timeout=18446744073709551615) at ../src/libsystemd/sd-event/sd-event.c:4236
#21 0x00007fc3c11bc389 in manager_loop (m=0x61c000000080) at ../src/core/manager.c:3068
#22 0x000000000041c11d in invoke_main_loop (m=0x61c000000080, saved_rlimit_nofile=0x7fc3b9f011d0, saved_rlimit_memlock=0x7fc3b9f011f0, ret_retval=0x7fc3b9f01040, ret_shutdown_verb=0x7fc3b9f01090,
    ret_fds=0x7fc3b9f010f0, ret_switch_root_dir=0x7fc3b9f01050, ret_switch_root_init=0x7fc3b9f01070, ret_error_message=0x7fc3b9f010b0) at ../src/core/main.c:1894
#23 0x000000000042477e in main (argc=3, argv=0x7ffcb71c4b68) at ../src/core/main.c:2957
```

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #23278.

### CODE REVIEW

**Comment by yuwata** [comment]
cc @evverx and @mrc0mmand

File: src/core/unit.c [hunk 1]
```diff
@@ -5842,6 +5842,9 @@ static int unit_freezer_action(Unit *u, FreezerAction action) {
         if (s != UNIT_ACTIVE)
                 return -EHOSTDOWN;
 
+        if (u->pending_freezer_message)
+                return -EALREADY;
+
```
Comment (msekletar) [inline src/core/unit.c]: I am just wondering why check that is immediately below is not enough.

Comment (yuwata) [reply src/core/unit.c]: Indeed. When pending_freezer_message is set, then the state must be FREEZING or THAWING.

**Comment by yuwata** [comment]
@msekletar Updated. Please take a look.

**Comment by evverx** [comment]
@yuwata thanks! I can longer trigger that crash with this patch applied.

FWIW @mrc0mmand I think it would be great if the `-x` option could affect void methods as well. As far as I can tell even with `-x 1000` `dfuzzer` calls void methods 10 times.

**Review by msekletar — APPROVED** [review]
LGTM

**Pull request merged**